zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/ZNodeGroupAclProperties.java

```diff
@@ -120,6 +130,19 @@ public static ZNodeGroupAclProperties getInstance() {
     return superUserDomainNames;
   }
 
+  public static boolean shouldSetX509ClientIdAsAcl() {
+    return QuorumPeerConfig.isSetX509ClientIdAsAclEnabled() && !ZNodeGroupAclProperties
+        .getInstance().isConnectionFilteringEnabled();
```

Comment:
I'm wondering why these two need to be mutually exclusive? Shouldn't a user be able to use the `isSetX509ClientIdAsAclEnabled` feature even though the server is dedicated?

Reply:
Good point. My original thought is that connect filtering feature would disable the setX509ClientIdAsAcl. But you are right, they don't need to be mutually exclusive.